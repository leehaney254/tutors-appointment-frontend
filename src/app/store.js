import { configureStore } from '@reduxjs/toolkit';
import registerReducer from '../features/RegisterSlice';
import loginReducer from '../features/LoginSlice';

export default configureStore({
  reducer: {
    register: registerReducer,
    login: loginReducer,
  },
});
