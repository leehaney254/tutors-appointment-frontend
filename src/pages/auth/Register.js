import React from 'react';

const Register = () => <div> Register page </div>;

export default Register;
