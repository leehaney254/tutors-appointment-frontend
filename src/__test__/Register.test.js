import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../app/store';
import Register from '../pages/auth/Register';

it('should render Register page', () => {
  const register = render(
    <BrowserRouter>
      <Provider store={store}>
        <Register />
      </Provider>
    </BrowserRouter>,
  );
  expect(register).toMatchSnapshot();
});
