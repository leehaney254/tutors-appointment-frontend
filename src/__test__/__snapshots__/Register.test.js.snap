// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Register page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative flex flex-col justify-center min-h-screen overflow-hidden"
      >
        <div
          class="w-full p-6 m-auto bg-white rounded-md shadow-xl shadow-rose-600/40 ring ring-2 ring-green-600 lg:max-w-xl"
        >
          <h3
            class="text-2xl font-semibold text-center text-green-500 underline uppercase decoration-wavy"
          >
            Registration form
          </h3>
          <form
            class="mt-6"
          >
            <div
              class="mb-2"
            >
              <label
                class="block text-sm font-semibold text-gray-800"
                for="name"
              >
                Username
              </label>
              <input
                class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
                id="name"
                name="name"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-2"
            >
              <label
                class="block text-sm font-semibold text-gray-800"
                for="password"
              >
                Password
              </label>
              <input
                class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="mb-2"
            >
              <label
                class="block text-sm font-semibold text-gray-800"
                for="cpassword"
              >
                Confirm Password
              </label>
              <input
                class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
                id="cpassword"
                name="cpassword"
                type="password"
                value=""
              />
            </div>
            <div
              class="mt-6"
            >
              <button
                class="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-green-500 rounded-md hover:bg-green-600 focus:outline-none focus:bg-green-600"
                type="submit"
              >
                Register Now
              </button>
            </div>
            <p
              class="mb-0 mt-2 pt-1 text-sm font-semibold"
            >
              You have an account?
              <a
                class="text-danger transition duration-150 ease-in-out hover:text-danger-600 focus:text-danger-600 active:text-danger-700"
                href="/login"
              >
                Login here
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative flex flex-col justify-center min-h-screen overflow-hidden"
    >
      <div
        class="w-full p-6 m-auto bg-white rounded-md shadow-xl shadow-rose-600/40 ring ring-2 ring-green-600 lg:max-w-xl"
      >
        <h3
          class="text-2xl font-semibold text-center text-green-500 underline uppercase decoration-wavy"
        >
          Registration form
        </h3>
        <form
          class="mt-6"
        >
          <div
            class="mb-2"
          >
            <label
              class="block text-sm font-semibold text-gray-800"
              for="name"
            >
              Username
            </label>
            <input
              class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
              id="name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-2"
          >
            <label
              class="block text-sm font-semibold text-gray-800"
              for="password"
            >
              Password
            </label>
            <input
              class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-2"
          >
            <label
              class="block text-sm font-semibold text-gray-800"
              for="cpassword"
            >
              Confirm Password
            </label>
            <input
              class="block w-full px-4 py-2 mt-2 text-slate-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
              id="cpassword"
              name="cpassword"
              type="password"
              value=""
            />
          </div>
          <div
            class="mt-6"
          >
            <button
              class="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-green-500 rounded-md hover:bg-green-600 focus:outline-none focus:bg-green-600"
              type="submit"
            >
              Register Now
            </button>
          </div>
          <p
            class="mb-0 mt-2 pt-1 text-sm font-semibold"
          >
            You have an account?
            <a
              class="text-danger transition duration-150 ease-in-out hover:text-danger-600 focus:text-danger-600 active:text-danger-700"
              href="/login"
            >
              Login here
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
